### DISCUSSION TITLE
ERROR: Failed building wheel for klayout

### DISCUSSION THREAD

**Discussion by ghost** [iqm-finland/KQCircuits#85] [Q&A]
I am trying to install KQCircuits for use outside the KLayout editor GUI on a Windows 11 machine equipped with Python 3.12. I've tried running

_python -m pip install -e klayout_package/python_

or

_pip-sync requirements.txt doc-requirements.txt test-requirements.txt_

to perform the installation, but I always get the error message shown below:

![Screenshot 2024-03-11 101204](https://github.com/iqm-finland/KQCircuits/assets/160117585/19e65dcb-af43-422e-b83f-c0094fefe3b9)

I've tried installing a few different versions of Microsoft Visual C++ (14.38 and 14.32), but that didn't help. Anyone has a tip on how to solve this issue? Many thanks.

**Comment by qpavsmi** [comment]
Hi! This is not a fix but a way to isolate the problem:

If you run `pip install klayout` does it show the same error? If yes, are you able to install older klayout starting from `pip install klayout==0.28.16` and going backwards until version `0.28.1`? You will know that your KLayout is working if you can run the python interpreter with `python` and then can create a box without errors:

```
>>> import klayout.db as pya
>>> pya.DBox(0,0,100,100)
(0,0;100,100)
>>> exit()
```

**Reply by ghost** [reply]
Hello! Thanks for the quick reply, much appreciated:) I've tried installing older versions of KLayout all the way down to 0.28.1, but I always get the same error. Got ay other ideas?

**Reply by qpavsmi** [reply]
Seems that KLayout itself is not playing well with your system unfortunately. I can't come up with further ideas on the spot, but I can recommend that you open a bug report on KLayout's github page: https://github.com/KLayout/klayout hopefully they'll be able to help faster than me

**Reply by ghost** [reply]
Thank you, I'll have a go at their page then:)

**Comment by qpavsmi** [comment]
Seems to be resolved according to https://github.com/KLayout/klayout/issues/1653

Closing
